h5glance/datatypes.py

```diff
@@ -0,0 +1,115 @@
+from functools import lru_cache
+from h5py import h5t
+import numpy as np
+from .utils import fmt_shape
+
+cset_names = {h5t.CSET_ASCII: 'ASCII', h5t.CSET_UTF8: 'UTF-8'}
+
+# h5py TypeIDs are not consistently hashable, so we can't use them as dict keys.
+# Instead, we look up by size, and then check equality with the standard types.
+@lru_cache()
+def int_types_by_size():
+    d = {}
+    for tc in np.typecodes['AllInteger']:
+        _add_typecode(tc, d)
+    return d
+
+@lru_cache()
+def float_types_by_size():
+    d = {}
+    for tc in np.typecodes['Float']:
+        _add_typecode(tc, d)
+    return d
+
+def _add_typecode(tc, sizes_dict):
+    dt_le = np.dtype('<' + tc)
+    dt_be = np.dtype('>' + tc)
+
+    entries = sizes_dict.setdefault(dt_le.itemsize, [])
+    entries.append((h5t.py_create(dt_le), dt_le.name))
+    entries.append((h5t.py_create(dt_be), dt_be.name + ' (big-endian)'))
+
+
+def fmt_dtype(hdf_dt):
+    """Get a (preferably short) string describing an HDF5 datatype"""
+    size = hdf_dt.get_size()
+
+    if isinstance(hdf_dt, h5t.TypeIntegerID):
+        # Check normal int & uint dtypes first
+        for candidate, descr in int_types_by_size().get(size, ()):
+            if hdf_dt == candidate:
+                return descr
+        un = 'un' if hdf_dt.get_sign() == h5t.SGN_NONE else ''
+        return "{}-byte {}signed integer".format(size, un)
+    elif isinstance(hdf_dt, h5t.TypeFloatID):
+        # Check normal float dtypes first
+        for candidate, descr in float_types_by_size().get(size, ()):
+            if hdf_dt == candidate:
+                return descr
+        return "custom {}-byte float".format(size)
+    elif isinstance(hdf_dt, h5t.TypeBitfieldID):
+        return "{}-byte bitfield"
```

Comment:
you forgot to format the number of bytes here (same for TypeOpaque)

Reply:
Thanks, well caught!